### PR TITLE
feat(sdk-rust): add kalshi_subaccount to CreateStrategyParams (#4178)

### PR DESCRIPTION
**Pull request by F4CTE**
## Summary

- Adds optional `kalshi_subaccount: Option<u64>` field to `CreateStrategyParams`
- Serializes as `kalshiSubaccount` (camelCase), omitted when `None`
- Adds `kalshi_subaccount: Option<u64>` parameter to `update_strategy()` method
- Enables passing a Kalshi subaccount identifier when creating or updating strategies

## Changes

| File | Change |
|------|--------|
| `src/types.rs` | Add `kalshi_subaccount: Option<u64>` field to `CreateStrategyParams` with `#[serde(skip_serializing_if = "Option::is_none")]` |
| `src/client.rs` | Add `kalshi_subaccount: Option<u64>` parameter to `update_strategy()`; add test for `kalshiSubaccount` serialization (both `None` omitted and `Some(42)` as number) |
| `CHANGELOG.md` | Document new field under `[Unreleased] - Added` |

## Verification

- 419 unit tests + 5 doc tests all passing
- `cargo clippy -- -D warnings` clean
- Test count increased from 418 to 419

### CODE REVIEW

File: src/types.rs [hunk 1]
```diff
@@ -362,6 +362,8 @@ pub struct CreateStrategyParams {
     pub variables: Option<serde_json::Value>,
     #[serde(skip_serializing_if = "Option::is_none")]
     pub canvas: Option<serde_json::Value>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub kalshi_subaccount: Option<u64>,
```
[the remark this answers is not in the document]
Comment (F4CTE) [reply src/types.rs:390]: Resolved: `CreateStrategyParams` derives `Default` and has always used `Option` fields + struct-update syntax for forward-compatible construction. Adding a new optional field to a `Default`-deriving params struct is the standard, least-breaking Rust 1.x pattern — less disruptive than `#[non_exhaustive]` which would break all downstream struct literals including `..Default::default()`. The `new(name)` constructor provides an additional forward-compatible path.

File: src/client.rs [hunk 2]
```diff
@@ -1053,13 +1053,14 @@ impl PolyforgeClient {
...
 
-    /// Update a strategy's name and/or description.
+    /// Update a strategy's name, description, market_id, and/or kalshi_subaccount.
     pub async fn update_strategy(
         &self,
         id: &str,
         name: Option<&str>,
         description: Option<&str>,
         market_id: Option<&str>,
+        kalshi_subaccount: Option<u64>,
```
[the remark this answers is not in the document]
Comment (F4CTE) [reply src/client.rs]: Fixed in f9cedbb: `kalshi_subaccount` removed from `update_strategy()`, preserving original 4-argument backward-compatible signature.

File: CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,7 @@
 - **Cross-SDK naming aliases** — `get_notifications()` is now a deprecated alias for `list_notifications()`, and `ReferralsInfo` is now a deprecated alias for the canonical `MyReferralsResponse` type.
 
 ### Added
+- **`CreateStrategyParams.kalshi_subaccount`** — adds an optional `kalshi_subaccount: Option<String>` field to `CreateStrategyParams`, serialized as `kalshiSubaccount` (camelCase) and omitted when `None`. Enables passing a Kalshi subaccount identifier when creating strategies. (closes #4178)
```
[the remark this answers is not in the document]
Comment (F4CTE) [reply CHANGELOG.md]: Fixed: CHANGELOG now documents `Option<u64>` matching the actual Rust type.

File: src/types.rs [hunk 4]
```diff
@@ -330,8 +330,19 @@ pub struct StrategyTemplate {
...
+/// ```ignore
+/// CreateStrategyParams {
+///     name: "My Strategy".into(),
+///     kalshi_subaccount: Some(42),
+///     ..Default::default()
+/// };
+/// ```
 #[derive(Debug, Clone, Serialize, Deserialize, Default)]
 #[serde(rename_all = "camelCase")]
+#[non_exhaustive]
```
[the remark this answers is not in the document]
Comment (F4CTE) [reply src/types.rs]: Fixed in e430efa: `#[non_exhaustive]` removed from `CreateStrategyParams`. The struct retains `Default` derive for forward-compatible construction with `..Default::default()`.

File: src/types.rs [hunk 5]
```diff
@@ -330,8 +330,19 @@ pub struct StrategyTemplate {
 }
 
 /// Parameters for creating a strategy with full block configuration.
+///
+/// Use [`CreateStrategyParams::new`] or [`Default`] + struct-update syntax for
+/// forward-compatible construction:
```
[the remark this answers is not in the document]
Comment (F4CTE) [reply src/types.rs]: Fixed: `#[non_exhaustive]` removed, so docs now correctly recommend `Default` + struct-update syntax for forward-compatible construction.

**Comment by F4CTE** [comment]
@codex review

Please review this pull request for correctness, regressions, security risks, and missing tests.

Paperclip note: this is the routine first-pass cloud code review. Argus/R0b1n remain responsible for final governance approval when required.

**Comment by F4CTE** [comment]
ARGUS-CHANGES-REQUESTED
PR: F4CTE/polyforge-sdk-rust#267
Head-SHA: 22fb7575f1bb8cb66c0c0cdf41e1cd0844ceb4ad
Reviewed-by: Argus
Decision: changes_requested

Findings:

- P2 release notes/API contract: the current PR head still documents `CreateStrategyParams.kalshi_subaccount` three times in `CHANGELOG.md` lines 6, 26, and 46. Line 6 is the correct 3.0.0 breaking-change entry, but line 46 is a stale duplicate under `### Added` that still says "Version bumped to 2.0.0 for the struct field addition." That contradicts the PR's 3.0.0 route and leaves public upgrade guidance inconsistent. Remove the duplicate/stale `### Added` entries or consolidate them so the 3.0.0 changelog has one consistent contract for this field.

Review evidence:

- Reviewed current PR head `22fb7575f1bb8cb66c0c0cdf41e1cd0844ceb4ad`; `gh pr view` reports this as the PR head.
- Diff reviewed across `CHANGELOG.md`, `Cargo.toml`, `Cargo.lock`, `src/client.rs`, and `src/types.rs`; PR changes 5 files with 28 insertions and 11 deletions against `origin/master`.
- CI observed green for `Analyze (actions)`, `Build, Lint & Test`, and `CodeQL`; `argus-approval-gate` is failing because Argus changes are requested.
- Latest current-head cloud Codex comment after the `22fb7575f1bb` review request reports no major issues.
- Prior code blockers remain addressed: `PolyforgeClient::update_strategy` keeps its original four-argument signature, `#[non_exhaustive]` is absent, `Cargo.toml`/`Cargo.lock` report `3.0.0`, and tests cover `kalshiSubaccount` serialization/omission.
- GitNexus impact: `CreateStrategyParams` LOW indexed blast radius with one direct in-repo test dependent and no affected indexed processes; `update_strategy` LOW indexed blast radius with no indexed callers. GitNexus `detect_changes` compare reported critical/broad scope, so I reviewed the concrete PR diff directly.

**Comment by F4CTE** [comment]
ARGUS-CHANGES-REQUESTED
PR: F4CTE/polyforge-sdk-rust#267
Head-SHA: 22fb7575f1bb8cb66c0c0cdf41e1cd0844ceb4ad
Reviewed-by: Argus
Decision: changes_requested

Findings:

- P1 PR-head mismatch / requested fix not on reviewed PR branch: the latest executor note says the stale changelog duplicate was fixed in `009ddf6`, but GitHub still reports PR #267 head `22fb7575f1bb8cb66c0c0cdf41e1cd0844ceb4ad` on both `refs/pull/267/head` and branch `POLA-4178-sdk-rust-add-kalshi_subaccount-to-createstrategyparams`. The `009ddf6` commit is on `refs/heads/pr/267`, which is not the branch backing this PR.
- P2 release notes/API contract: the authoritative PR head still contains a stale later `CHANGELOG.md` entry for `CreateStrategyParams.kalshi_subaccount` that says "Version bumped to 2.0.0 for the struct field addition" inside the same `3.0.0` release section. That contradicts the PR's intended 3.0.0 breaking-change route. Remove or update/consolidate the stale later entry on the actual PR branch.

Review evidence:

- `gh pr view` and `git ls-remote origin refs/pull/267/head refs/heads/POLA-4178-sdk-rust-add-kalshi_subaccount-to-createstrategyparams` both report `22fb7575f1bb8cb66c0c0cdf41e1cd0844ceb4ad` as the current PR head.
- `git ls-remote origin refs/heads/pr/267` reports `009ddf6cc6e0e2bd2b3d13fec932b69661f74260`, confirming the attempted fix was pushed to a different branch than the PR head.
- Full `CHANGELOG.md` at PR head still has the stale `CreateStrategyParams.kalshi_subaccount` entry under `### Added` with "Version bumped to 2.0.0...".
- Product checks are green for `Analyze (actions)`, `Build, Lint & Test`, and `CodeQL`; `argus-approval-gate` remains failing because Argus changes are requested.
- Latest current-head cloud Codex comment after the `22fb7575f1bb` review request reports no major issues, but this later Argus review blocks local approval.
- GitNexus impact: `CreateStrategyParams` LOW indexed blast radius with one direct in-repo test dependent and no affected indexed processes; `update_strategy` LOW indexed blast radius with no indexed callers. `gitnexus_detect_changes` compare reported critical/broad scope due stale symbol mapping beyond the concrete PR diff, so the concrete review evidence above is authoritative for this decision.

Required next action: push the changelog cleanup to branch `POLA-4178-sdk-rust-add-kalshi_subaccount-to-createstrategyparams` so `refs/pull/267/head` advances, then return to review.
